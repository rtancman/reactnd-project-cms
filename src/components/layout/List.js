import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from "react-router-dom"
import moment from 'moment'
import sortBy from 'sort-by'
import CommentIcon from '@material-ui/icons/Comment';
import EventIcon from '@material-ui/icons/Event';
import './List.css'
import Vote from '../vote'

const orderByOptions = [
  {label: 'Popular', value: '-voteScore'},
  {label: 'Category', value: 'category'},
  {label: 'Recent', value: '-timestamp'},
]

class BaseList extends Component {
  static propTypes = {
    items: PropTypes.array.isRequired,
    title: PropTypes.string,
  }

  state = {
    orderBy: false,
  }

  orderBy(option) {
    this.setState({
      orderBy: option
    })
  }
}

class ListContent extends BaseList {

  state = {
    orderBy: '-voteScore'
  }

  render() {
    const { items, title } = this.props
    const { orderBy } = this.state
    let showingPosts = items

    showingPosts.sort(sortBy(orderBy))

    return (
      <div className="list_content">
        <h2>{ title }</h2>
        <div className="row list_content__filter">
          { orderByOptions.map((option) => (
            <div data-filter-option-value={ option.value } onClick={ () => this.orderBy(option.value)} key={ option.value } className={ `col-xs list_content__filter__item ${option.value === orderBy? 'list_content__filter__item--selected' : ''}` }>
              <span>{ option.label }</span>
            </div>
          )) }
        </div>
        {showingPosts.map((item) => (
          <div className="list_content__item" key={item.id}>
            <div className="list_content__item__title">
              <Link className="link" to={`/${item.category}/${item.id}`}>{item.title}</Link>
              <div className="list_content__item__title--vote">
                <Vote
                  type='post' 
                  id={item.id}
                  total={item.voteScore || 0}
                />
              </div>
            </div>
            <div className="list_content__item__info">
              <p>
                <EventIcon style={{ fontSize: 10 }} title='Create at' alt='Create at' /> { moment(item.timestamp).format('MM/DD/YYYY') } <CommentIcon style={{ fontSize: 10 }} title='Total comments' alt='Total comments' /> { item.commentCount } - in <Link className="link" to={`/${item.category}`}>{ item.category }</Link> - by <span className="list_posts__item__info--author">{ item.author }</span>
              </p>
            </div>
          </div>
        ))}
      </div>
    )
  }
}

const orderByCategoryOptions = [
  {label: 'Default', value: false},
  {label: 'Name', value: 'path'},
]
class List extends BaseList {
  render() {
    const { items, title } = this.props
    const { orderBy } = this.state
    let showingItems = items
    if ( orderBy ){
      showingItems.sort(sortBy(orderBy))
    }

    return (
      <div className="list_content">
        <h2>{ title }</h2>
        <div className="row list_content__filter">
          { orderByCategoryOptions.map((option) => (
            <div data-filter-option-value={ option.value } onClick={ () => this.orderBy(option.value)} key={ option.value } className={ `col-xs list_content__filter__item ${option.value === orderBy? 'list_content__filter__item--selected' : ''}` }>
              <span>{ option.label }</span>
            </div>
          )) }
        </div>
        {showingItems.map((item) => (
          <div className="list_content__item list_content__item--category" key={item.path}>
            <div className="list_content__item__title">
              <Link className="link" to={`/${item.path}`}>{item.name}</Link>
            </div>
          </div>
        ))}
      </div>
    )
  }
}

List.propTypes = {
  items: PropTypes.array.isRequired,
}

export { ListContent, List }