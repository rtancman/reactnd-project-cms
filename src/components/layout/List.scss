@import '../_mediaqueries.scss';

.list_content {
  &__item {
    margin-top: 30px;
    margin-bottom: 10px;
    padding-bottom: 20px;
    border-bottom: 0.5px solid #E1E1E1;
    
    &:last-child {
      border-bottom: 0;
    }

    &--category {
      padding-bottom: 37px;
    }

    &__title {
      font-size: 24px;
      letter-spacing: -1.25px;
      line-height: 28px;
      font-weight: 600;
      display: flex;

      @include small {
        flex-direction: column;
      }

      a {
        flex: 1 1;
        @include small {
          margin-bottom: 10px;
        }
      }
      &--vote {
        justify-content: flex-end;
        font-size: 17px;
        .vote__bar__total {
          padding: 0 10px;
        }
        @include small {
          flex: 1 1;
          margin-bottom: 10px;
        }
      }
    }
  
    &__info {
      color: #999;
      font-size: 11px;
      grid-column: 2;
      line-height: 11px;
      margin: 6px 0 0;
      text-transform: uppercase;
      p {
        margin: 0;
      }
    }
  }

  &__filter {
    margin: 0;

    &__item {
      padding: 12px 0 9px 0;
      border: 0.5px solid #e0e0e0;
      font-weight: bold;
      font-size: 12px;
      color: #666;
      text-transform: uppercase;
      text-align: center;
      cursor: pointer;
      line-height: 1;

      &:first-child {
        border-radius: 3px 0 0 3px;
      }

      &:last-child {
        border-radius: 0 3px 3px 0;
      }

      &--selected {
        border-color: #000;
        color: #000;
      }
    }
  }
}