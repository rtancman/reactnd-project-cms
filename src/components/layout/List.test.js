import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router'
import { List, ListContent } from './List'

describe('List Component', () => {
  describe('when component renders', () => {
    it('without crashing', () => {
      const props = {
        items: [],
        title: 'Title list',
      }
      const wrapper = shallow(<List {...props} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('with valid props display list content', () => {
      const props = {
        items: [
          { path: '4', name: 'item 4', },
        ],
        title: 'Title list',
      }
      const wrapper = mount(
        <MemoryRouter>
          <List {...props} />
        </MemoryRouter>
      )

      expect(wrapper.find('.list_content h2').length).toBe(1)
      expect(wrapper.find('.list_content h2').text()).toBe('Title list')
      expect(wrapper.find('.list_content__item').length).toBe(1)
      const item = wrapper.find('.list_content__item').first()
      expect(item.text()).toBe('item 4')
      expect(item.find('Link').length).toBe(1)
      expect(item.find('Link').instance().props.to).toBe('/4')
    })
  })

  describe('#orderBy', () => {
    describe('when user does not click', () => {
      it('render list like props.items', () => {
        const props = {
          items: [
            { path: '4', name: 'item 4', },
            { path: '1', name: 'item 1', },
            { path: '5', name: 'item 5', },
            { path: '3', name: 'item 3', },
            { path: '2', name: 'item 2', },
          ],
          title: 'Title list',
        }
        const wrapper = mount(
          <MemoryRouter>
            <List {...props} />
          </MemoryRouter>
        )
  
        expect(wrapper.find('List').props().items).toEqual(props.items)
      })
    })

    describe('when user click', () => {
      it('in option Default render list like props.items', () => {
        const props = {
          items: [
            { path: '4', name: 'item 4', },
            { path: '1', name: 'item 1', },
            { path: '5', name: 'item 5', },
            { path: '3', name: 'item 3', },
            { path: '2', name: 'item 2', },
          ],
          title: 'Title list',
        }
        const wrapper = mount(
          <MemoryRouter>
            <List {...props} />
          </MemoryRouter>
        )
  
        wrapper.find('div.list_content__filter__item[data-filter-option-value=false]').simulate('click')
        expect(wrapper.find('List').props().items).toEqual(props.items)
        expect(wrapper.find('List').instance().state.orderBy).toEqual(false)
        const item = wrapper.find('.list_content__item').first()
        expect(item.text()).toBe('item 4')
        expect(item.find('Link').length).toBe(1)
        expect(item.find('Link').instance().props.to).toBe('/4')
      })
  
      it('in option Name render list sorted by path', () => {
        const sortedByPath = [
          { path: '1', name: 'item 1', },
          { path: '2', name: 'item 2', },
          { path: '3', name: 'item 3', },
          { path: '4', name: 'item 4', },
          { path: '5', name: 'item 5', },
        ]
        const props = {
          items: [
            { path: '4', name: 'item 4', },
            { path: '1', name: 'item 1', },
            { path: '5', name: 'item 5', },
            { path: '3', name: 'item 3', },
            { path: '2', name: 'item 2', },
          ],
          title: 'Title list',
        }
        const wrapper = mount(
          <MemoryRouter>
            <List {...props} />
          </MemoryRouter>
        )
  
        wrapper.find('div.list_content__filter__item[data-filter-option-value="path"]').simulate('click')
        expect(wrapper.find('List').props().items).toEqual(sortedByPath)
        expect(wrapper.find('List').instance().state.orderBy).toEqual('path')
        const item = wrapper.find('.list_content__item').first()
        expect(item.text()).toBe('item 1')
        expect(item.find('Link').length).toBe(1)
        expect(item.find('Link').instance().props.to).toBe('/1')
      })
    })
  })
})

describe('ListContent Component', () => {
  describe('when component renders', () => {
    it('without crashing', () => {
      const props = {
        items: [],
        title: 'Title list content',
      }
      const wrapper = shallow(<ListContent {...props} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('with valid props display list content', () => {
      const props = {
        items: [
          {
            "id": "8xf0y6ziyjabvozdd253nd",
            "timestamp": 1467166872634,
            "title": "Udacity is the best place to learn React",
            "body": "Everyone says so after all.",
            "author": "thingtwo",
            "category": "react",
            "voteScore": 6,
            "deleted": false,
            "commentCount": 2
          },
        ],
        title: 'Title list content',
      }
      const wrapper = mount(
        <MemoryRouter>
          <ListContent {...props} />
        </MemoryRouter>
      )

      expect(wrapper.find('.list_content h2').length).toBe(1)
      expect(wrapper.find('.list_content h2').text()).toBe('Title list content')
      expect(wrapper.find('.list_content__item').length).toBe(1)
      const item = wrapper.find('.list_content__item').first()
      const itemTitle = item.find('.list_content__item__title')
      expect(itemTitle.text()).toBe('Udacity is the best place to learn React')
      expect(itemTitle.find('Link').length).toBe(1)
      expect(itemTitle.find('Link').instance().props.to).toBe('/react/8xf0y6ziyjabvozdd253nd')
      const itemInfo = item.find('.list_content__item__info')
      expect(itemInfo.text()).toBe('date 06-28-2016 - in react - by thingtwo')
      expect(itemInfo.find('Link').length).toBe(1)
      expect(itemInfo.find('Link').instance().props.to).toBe('/react')
    })
  })

  describe('#orderBy', () => {
    describe('when user does not click', () => {
      it('render list sorted by default orderBy -voteScore', () => {
        const props = {
          items: [
            {
              "id": "8xf0y6ziyjabvozdd253nd",
              "timestamp": 1467166872634,
              "title": "Udacity is the best place to learn React",
              "body": "Everyone says so after all.",
              "author": "thingtwo",
              "category": "react",
              "voteScore": 6,
              "deleted": false,
              "commentCount": 2
            },
            {
              "id": "6ni6ok3ym7mf1p33lnez",
              "timestamp": 1468479767190,
              "title": "Learn Redux in 10 minutes!",
              "body": "Just kidding. It takes more than 10 minutes to learn technology.",
              "author": "lala",
              "category": "redux",
              "voteScore": 105,
              "deleted": false,
              "commentCount": 0
            }
          ],
          title: 'Title list content',
        }
        const wrapper = mount(
          <MemoryRouter>
            <ListContent {...props} />
          </MemoryRouter>
        )
  
        expect(wrapper.find('ListContent').instance().state.orderBy).toEqual('-voteScore')
        const item = wrapper.find('.list_content__item').first()
        const itemTitle = item.find('.list_content__item__title')
        expect(itemTitle.text()).toBe('Learn Redux in 10 minutes!')
        expect(itemTitle.find('Link').length).toBe(1)
        expect(itemTitle.find('Link').instance().props.to).toBe('/redux/6ni6ok3ym7mf1p33lnez')
        const itemInfo = item.find('.list_content__item__info')
        expect(itemInfo.text()).toBe('date 07-14-2016 - in redux - by lala')
        expect(itemInfo.find('Link').length).toBe(1)
        expect(itemInfo.find('Link').instance().props.to).toBe('/redux')
      })
    })

    describe('when user click', () => {
      it('in option Category render list sorted by post category', () => {
        const props = {
          items: [
            {
              "id": "8xf0y6ziyjabvozdd253nd",
              "timestamp": 1467166872634,
              "title": "Udacity is the best place to learn React",
              "body": "Everyone says so after all.",
              "author": "thingtwo",
              "category": "react",
              "voteScore": 6,
              "deleted": false,
              "commentCount": 2
            },
            {
              "id": "6ni6ok3ym7mf1p33lnez",
              "timestamp": 1468479767190,
              "title": "Learn Redux in 10 minutes!",
              "body": "Just kidding. It takes more than 10 minutes to learn technology.",
              "author": "lala",
              "category": "redux",
              "voteScore": 105,
              "deleted": false,
              "commentCount": 0
            },
          ],
          title: 'Title list',
        }
        const expected = [
          {title: 'Udacity is the best place to learn React', linkTitle: '/react/8xf0y6ziyjabvozdd253nd', info: 'date 06-28-2016 - in react - by thingtwo', linkInfo: '/react'},
          {title: 'Learn Redux in 10 minutes!', linkTitle: '/redux/6ni6ok3ym7mf1p33lnez', info: 'date 07-14-2016 - in redux - by lala', linkInfo: '/redux'},
        ]

        const wrapper = mount(
          <MemoryRouter>
            <ListContent {...props} />
          </MemoryRouter>
        )
  
        wrapper.find('div.list_content__filter__item[data-filter-option-value="category"]').simulate('click')
        expect(wrapper.find('ListContent').instance().state.orderBy).toEqual('category')
        wrapper.find('.list_content__item').forEach((node, i) => {
          let itemTitle = node.find('.list_content__item__title')
          expect(itemTitle.text()).toBe(expected[i].title)
          expect(itemTitle.find('Link').length).toBe(1)
          expect(itemTitle.find('Link').instance().props.to).toBe(expected[i].linkTitle)
          let itemInfo = node.find('.list_content__item__info')
          expect(itemInfo.text()).toBe(expected[i].info)
          expect(itemInfo.find('Link').length).toBe(1)
          expect(itemInfo.find('Link').instance().props.to).toBe(expected[i].linkInfo)
        });
      })
  
      it('in option Recent render list sorted by post -timestamp', () => {
        const props = {
          items: [
            {
              "id": "8xf0y6ziyjabvozdd253nd",
              "timestamp": 1468479767190,
              "title": "Udacity is the best place to learn React",
              "body": "Everyone says so after all.",
              "author": "thingtwo",
              "category": "react",
              "voteScore": 6,
              "deleted": false,
              "commentCount": 2
            },
            {
              "id": "1468479767190",
              "timestamp": 1539053910610,
              "title": "teste 3",
              "body": "teste body 3",
              "author": "lele",
              "category": "oieee",
              "voteScore": 6,
              "deleted": false,
              "commentCount": 2
            },
            {
              "id": "6ni6ok3ym7mf1p33lnez",
              "timestamp": 1539053766412,
              "title": "Learn Redux in 10 minutes!",
              "body": "Just kidding. It takes more than 10 minutes to learn technology.",
              "author": "lala",
              "category": "redux",
              "voteScore": 105,
              "deleted": false,
              "commentCount": 0
            },
          ],
          title: 'Title list',
        }
        const expected = [
          {title: 'teste 3', linkTitle: '/oieee/1468479767190', info: 'date 10-08-2018 - in oieee - by lele', linkInfo: '/oieee'},
          {title: 'Learn Redux in 10 minutes!', linkTitle: '/redux/6ni6ok3ym7mf1p33lnez', info: 'date 10-08-2018 - in redux - by lala', linkInfo: '/redux'},
          {title: 'Udacity is the best place to learn React', linkTitle: '/react/8xf0y6ziyjabvozdd253nd', info: 'date 07-14-2016 - in react - by thingtwo', linkInfo: '/react'},
        ]
        const wrapper = mount(
          <MemoryRouter>
            <ListContent {...props} />
          </MemoryRouter>
        )

        wrapper.find('div.list_content__filter__item[data-filter-option-value="-timestamp"]').simulate('click')
        expect(wrapper.find('ListContent').instance().state.orderBy).toEqual('-timestamp')
        wrapper.find('.list_content__item').forEach((node, i) => {
          let itemTitle = node.find('.list_content__item__title')
          expect(itemTitle.text()).toBe(expected[i].title)
          expect(itemTitle.find('Link').length).toBe(1)
          expect(itemTitle.find('Link').instance().props.to).toBe(expected[i].linkTitle)
          let itemInfo = node.find('.list_content__item__info')
          expect(itemInfo.text()).toBe(expected[i].info)
          expect(itemInfo.find('Link').length).toBe(1)
          expect(itemInfo.find('Link').instance().props.to).toBe(expected[i].linkInfo)
        });
      })

      it('in option Popular render list sorted by post -voteScore', () => {
        const props = {
          items: [
            {
              "id": "8xf0y6ziyjabvozdd253nd",
              "timestamp": 1468479767190,
              "title": "Udacity is the best place to learn React",
              "body": "Everyone says so after all.",
              "author": "thingtwo",
              "category": "react",
              "voteScore": 611,
              "deleted": false,
              "commentCount": 2
            },
            {
              "id": "1468479767190",
              "timestamp": 1539053910610,
              "title": "teste 3",
              "body": "teste body 3",
              "author": "lele",
              "category": "oieee",
              "voteScore": 42,
              "deleted": false,
              "commentCount": 2
            },
            {
              "id": "6ni6ok3ym7mf1p33lnez",
              "timestamp": 1539053766412,
              "title": "Learn Redux in 10 minutes!",
              "body": "Just kidding. It takes more than 10 minutes to learn technology.",
              "author": "lala",
              "category": "redux",
              "voteScore": 1005,
              "deleted": false,
              "commentCount": 0
            },
          ],
          title: 'Title list',
        }
        const expected = [
          {title: 'Learn Redux in 10 minutes!', linkTitle: '/redux/6ni6ok3ym7mf1p33lnez', info: 'date 10-08-2018 - in redux - by lala', linkInfo: '/redux'},
          {title: 'Udacity is the best place to learn React', linkTitle: '/react/8xf0y6ziyjabvozdd253nd', info: 'date 07-14-2016 - in react - by thingtwo', linkInfo: '/react'},
          {title: 'teste 3', linkTitle: '/oieee/1468479767190', info: 'date 10-08-2018 - in oieee - by lele', linkInfo: '/oieee'},
        ]

        const wrapper = mount(
          <MemoryRouter>
            <ListContent {...props} />
          </MemoryRouter>
        )

        wrapper.find('div.list_content__filter__item[data-filter-option-value="-voteScore"]').simulate('click')
        expect(wrapper.find('ListContent').instance().state.orderBy).toEqual('-voteScore')
        wrapper.find('.list_content__item').forEach((node, i) => {
          let itemTitle = node.find('.list_content__item__title')
          expect(itemTitle.text()).toBe(expected[i].title)
          expect(itemTitle.find('Link').length).toBe(1)
          expect(itemTitle.find('Link').instance().props.to).toBe(expected[i].linkTitle)
          let itemInfo = node.find('.list_content__item__info')
          expect(itemInfo.text()).toBe(expected[i].info)
          expect(itemInfo.find('Link').length).toBe(1)
          expect(itemInfo.find('Link').instance().props.to).toBe(expected[i].linkInfo)
        });
      })
    })
  })
})