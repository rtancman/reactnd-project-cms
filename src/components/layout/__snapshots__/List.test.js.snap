// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component when component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    items={Array []}
    title="Title list"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Title list
</h2>,
        <div
          className="row list_content__filter"
>
          <div
                    className="col-xs list_content__filter__item list_content__filter__item--selected"
                    data-filter-option-value={false}
                    onClick={[Function]}
          >
                    <span>
                              Default
                    </span>
          </div>
          <div
                    className="col-xs list_content__filter__item "
                    data-filter-option-value="path"
                    onClick={[Function]}
          >
                    <span>
                              Name
                    </span>
          </div>
</div>,
        Array [],
      ],
      "className": "list_content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title list",
        },
        "ref": null,
        "rendered": "Title list",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-xs list_content__filter__item list_content__filter__item--selected"
              data-filter-option-value={false}
              onClick={[Function]}
>
              <span>
                            Default
              </span>
</div>,
            <div
              className="col-xs list_content__filter__item "
              data-filter-option-value="path"
              onClick={[Function]}
>
              <span>
                            Name
              </span>
</div>,
          ],
          "className": "row list_content__filter",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "false",
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Default
</span>,
              "className": "col-xs list_content__filter__item list_content__filter__item--selected",
              "data-filter-option-value": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Default",
              },
              "ref": null,
              "rendered": "Default",
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "path",
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Name
</span>,
              "className": "col-xs list_content__filter__item ",
              "data-filter-option-value": "path",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name",
              },
              "ref": null,
              "rendered": "Name",
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Title list
</h2>,
          <div
            className="row list_content__filter"
>
            <div
                        className="col-xs list_content__filter__item list_content__filter__item--selected"
                        data-filter-option-value={false}
                        onClick={[Function]}
            >
                        <span>
                                    Default
                        </span>
            </div>
            <div
                        className="col-xs list_content__filter__item "
                        data-filter-option-value="path"
                        onClick={[Function]}
            >
                        <span>
                                    Name
                        </span>
            </div>
</div>,
          Array [],
        ],
        "className": "list_content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title list",
          },
          "ref": null,
          "rendered": "Title list",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-xs list_content__filter__item list_content__filter__item--selected"
                data-filter-option-value={false}
                onClick={[Function]}
>
                <span>
                                Default
                </span>
</div>,
              <div
                className="col-xs list_content__filter__item "
                data-filter-option-value="path"
                onClick={[Function]}
>
                <span>
                                Name
                </span>
</div>,
            ],
            "className": "row list_content__filter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "false",
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Default
</span>,
                "className": "col-xs list_content__filter__item list_content__filter__item--selected",
                "data-filter-option-value": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Default",
                },
                "ref": null,
                "rendered": "Default",
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "path",
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Name
</span>,
                "className": "col-xs list_content__filter__item ",
                "data-filter-option-value": "path",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Name",
                },
                "ref": null,
                "rendered": "Name",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ListContent Component when component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListContent
    items={Array []}
    title="Title list content"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Title list content
</h2>,
        <div
          className="row list_content__filter"
>
          <div
                    className="col-xs list_content__filter__item list_content__filter__item--selected"
                    data-filter-option-value="-voteScore"
                    onClick={[Function]}
          >
                    <span>
                              Popular
                    </span>
          </div>
          <div
                    className="col-xs list_content__filter__item "
                    data-filter-option-value="category"
                    onClick={[Function]}
          >
                    <span>
                              Category
                    </span>
          </div>
          <div
                    className="col-xs list_content__filter__item "
                    data-filter-option-value="-timestamp"
                    onClick={[Function]}
          >
                    <span>
                              Recent
                    </span>
          </div>
</div>,
        Array [],
      ],
      "className": "list_content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title list content",
        },
        "ref": null,
        "rendered": "Title list content",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-xs list_content__filter__item list_content__filter__item--selected"
              data-filter-option-value="-voteScore"
              onClick={[Function]}
>
              <span>
                            Popular
              </span>
</div>,
            <div
              className="col-xs list_content__filter__item "
              data-filter-option-value="category"
              onClick={[Function]}
>
              <span>
                            Category
              </span>
</div>,
            <div
              className="col-xs list_content__filter__item "
              data-filter-option-value="-timestamp"
              onClick={[Function]}
>
              <span>
                            Recent
              </span>
</div>,
          ],
          "className": "row list_content__filter",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "-voteScore",
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Popular
</span>,
              "className": "col-xs list_content__filter__item list_content__filter__item--selected",
              "data-filter-option-value": "-voteScore",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Popular",
              },
              "ref": null,
              "rendered": "Popular",
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "category",
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Category
</span>,
              "className": "col-xs list_content__filter__item ",
              "data-filter-option-value": "category",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Category",
              },
              "ref": null,
              "rendered": "Category",
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "-timestamp",
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Recent
</span>,
              "className": "col-xs list_content__filter__item ",
              "data-filter-option-value": "-timestamp",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Recent",
              },
              "ref": null,
              "rendered": "Recent",
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Title list content
</h2>,
          <div
            className="row list_content__filter"
>
            <div
                        className="col-xs list_content__filter__item list_content__filter__item--selected"
                        data-filter-option-value="-voteScore"
                        onClick={[Function]}
            >
                        <span>
                                    Popular
                        </span>
            </div>
            <div
                        className="col-xs list_content__filter__item "
                        data-filter-option-value="category"
                        onClick={[Function]}
            >
                        <span>
                                    Category
                        </span>
            </div>
            <div
                        className="col-xs list_content__filter__item "
                        data-filter-option-value="-timestamp"
                        onClick={[Function]}
            >
                        <span>
                                    Recent
                        </span>
            </div>
</div>,
          Array [],
        ],
        "className": "list_content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title list content",
          },
          "ref": null,
          "rendered": "Title list content",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-xs list_content__filter__item list_content__filter__item--selected"
                data-filter-option-value="-voteScore"
                onClick={[Function]}
>
                <span>
                                Popular
                </span>
</div>,
              <div
                className="col-xs list_content__filter__item "
                data-filter-option-value="category"
                onClick={[Function]}
>
                <span>
                                Category
                </span>
</div>,
              <div
                className="col-xs list_content__filter__item "
                data-filter-option-value="-timestamp"
                onClick={[Function]}
>
                <span>
                                Recent
                </span>
</div>,
            ],
            "className": "row list_content__filter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "-voteScore",
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Popular
</span>,
                "className": "col-xs list_content__filter__item list_content__filter__item--selected",
                "data-filter-option-value": "-voteScore",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Popular",
                },
                "ref": null,
                "rendered": "Popular",
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "category",
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Category
</span>,
                "className": "col-xs list_content__filter__item ",
                "data-filter-option-value": "category",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Category",
                },
                "ref": null,
                "rendered": "Category",
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "-timestamp",
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Recent
</span>,
                "className": "col-xs list_content__filter__item ",
                "data-filter-option-value": "-timestamp",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Recent",
                },
                "ref": null,
                "rendered": "Recent",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
